frontend/src/index.tsx

```diff
@@ -39,4 +42,16 @@ Date.prototype.toLocaleStringWithConditionalDate = function () {
 document.body.classList.add(cssVariables.style);
 document.body.classList.add(globalStyle);
 
-ReactDOM.render(<App />, document.getElementById('root') as HTMLElement);
+setRouter([
+  {
+    element: <App />,
+    children: pathRoutes
+  }
+]);
+
+// redirect to the router basename if the pathname does not include it
+if (!window.location.pathname.includes(rootBasename)) {
+  router.navigate('/', { replace: true });
+}
```

Comment:
voodoo.

Reply:
Here you define the router application:
- the parent element will be the `App` component (the one that will always be loaded). 
- the children paths are the components loaded depending on the URL (within the `<Outlet />` component used in `RenderPage.tsx`).

Lines 52-55 add a feature provided by React Router v5, but not by v6. The scenario is that if you add a basename for your router (in our case `/console`), this is not added automatically when you navigate to the root url (e.g. http://localhost:20001/kiali). Therefore, we have to force the navigation to `/console` if this basename is not included in the url pathname.

More info here: https://www.github.com/remix-run/react-router/issues/8427